### PR TITLE
Fix JavaScript parsing error in IE 10 and earlier

### PR DESCRIPTION
**Pull request by mramato**
While Cesium doesn't work in IE10, we need to allow the JS to parse so that an app can still function without Cesium.  In this case, Uint8ClampedArray isn't supported in IE 10, so the arraySlice polyfill was throwing on load. I cleaned up `arraySlice` and deduplicated common TypedArray code into FeatureDetection.

To Test, load Viewer in IE11 with the debugger open and set to IE10 compatibility mode.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@mramato, thanks for the pull request! Maintainers, we have a signed CLA from @mramato, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: Source/Core/arraySlice.js [hunk 1]
```diff
@@ -20,54 +30,22 @@ define([
...
-                    }
-                }
+        if (typeof array.slice === 'function') {
+            return array.slice(begin, end);
+        }
 
-                return copy;
-            };
+        var copy = Array.prototype.slice.call(array, begin, end);
+        var typedArrayTypes = FeatureDetection.typedarrayTypes;
```
Comment (hpinkos) [inline Source/Core/arraySlice.js]: `typearrayTypes` -> `typedArrayTypes`?

**Comment by hpinkos** [comment]
That's it, this change looks fine

**Comment by hpinkos** [comment]
👍

**Pull request merged**